.github/workflows/build-test.yml

```diff
@@ -19,43 +19,104 @@ jobs:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      examples: ${{ steps.filter.outputs.changes }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            java-chatroom: play-java-chatroom-example/**
+            java-compile-di: play-java-compile-di-example/**
+            java-dagger2: play-java-dagger2-example/**
+            java-ebean: play-java-ebean-example/**
+            java-fileupload: play-java-fileupload-example/**
+            java-forms: play-java-forms-example/**
+            java-grpc: play-java-grpc-example/**
+            java-hello-world: play-java-hello-world-tutorial/**
+            java-jpa: play-java-jpa-example/**
+            java-pekko-cluster: play-java-pekko-cluster-example/**
+            java-rest-api: play-java-rest-api-example/**
+            java-starter: play-java-starter-example/**
+            java-streaming: play-java-streaming-example/**
+            java-websocket: play-java-websocket-example/**
+            scala-anorm: play-scala-anorm-example/**
+            scala-chatroom: play-scala-chatroom-example/**
+            scala-compile-di: play-scala-compile-di-example/**
+            scala-fileupload: play-scala-fileupload-example/**
+            scala-forms: play-scala-forms-example/**
+            scala-grpc: play-scala-grpc-example/**
+            scala-hello-world: play-scala-hello-world-tutorial/**
+            scala-isolated-slick: play-scala-isolated-slick-example/**
+            scala-log4j2: play-scala-log4j2-example/**
+            scala-macwire-di: play-scala-macwire-di-example/**
+            scala-rest-api: play-scala-rest-api-example/**
+            scala-secure-session: play-scala-secure-session-example/**
+            scala-slick: play-scala-slick-example/**
+            scala-starter: play-scala-starter-example/**
+            scala-streaming: play-scala-streaming-example/**
+            scala-tls: play-scala-tls-example/**
+            scala-websocket: play-scala-websocket-example/**
 
-  gradle-tests:
-    name: Gradle Tests
-    uses: playframework/.github/.github/workflows/cmd.yml@v3
+  java-chatroom:
+    name: "Java :: Chatroom"
+    uses: playframework/.github/.github/workflows/cmd.yml@v4
+    needs: changes
+    if: ${{ (github.event_name != 'pull_request' && !failure() && !cancelled()) || contains(fromJSON(needs.changes.outputs.examples), 'java-chatroom')}}
     with:
-      java: 17, 11
+      java: 17, 21
       scala: 2.13.x, 3.x
+      gradle-build-root: play-java-chatroom-example
+      add-dimensions: >-
+        { "server": [ "pekko", "netty"] }
       cmd: |
-        ./test-gradle.sh
+        ./test.sh --sample=play-java-chatroom-example --backend-server=$MATRIX_SERVER
+        ./test.sh --sample=play-java-chatroom-example --backend-server=$MATRIX_SERVER --build-tool=gradle
```

Comment:
Does it make sense instead of two lines here make the buld-tool also part of the matrix so it's just one line:
```sh
./test.sh --sample=play-java-chatroom-example --backend-server=$MATRIX_SERVER --build-tool=$MATRIX_BUILD_TOOL
```
?

Reply:
Yes, of course, I also thought about that 👍